### ISSUE TITLE
Missing arity linting within extend-type or extend-protocol bodies

### ISSUE THREAD

**Issue by lassemaatta** [clj-kondo/clj-kondo#1333]
**version**

`clj-kondo v2021.06.18`

**macro usage**

Relates to both `clojure.core/extend-type` and `clojure.core/extend-protocol`.

**platform**

Native, Linux `5.13.4-arch2-1`.

**problem**

Calling a function with incorrect arity within the body of either `extend-type` or `extend-protocol` produces no errors.

**repro**

```clojure
(ns first-ns)

(defn a-function
  "A simple function with one required arg"
  [arg]
  arg)

(defprotocol SomeProtocol
  (foobar [this]))

(defprotocol AnotherProtocol
  (baz [this]))

(defprotocol ThirdProtocol
  (zed [this]))

(extend java.lang.Number
  SomeProtocol
  {:foobar (fn [_this]
             (a-function))}) ; -> An arity error, hurrah!

(defrecord SomeRecord [afield]
  SomeProtocol
  (foobar [_this]
    (a-function))) ; -> An arity error, hurrah!

(extend-type SomeRecord
  ThirdProtocol
  (zed [this]
    (let [x 2] ; -> An unused binding warning, hurrah!
     (a-function)))) ; -> No error :(

(extend-protocol AnotherProtocol
  SomeRecord
  (baz [this]
    (let [y 2] ; -> An unused binding warning, hurrah!
     (a-function)))) ; -> No error :(
```

Output

```shell
[lassemaatta@desktop clj-testing]$ clj-kondo --lint src/first_ns.clj
src/first_ns.clj:18:14: error: first-ns/a-function is called with 0 args but expects 1
src/first_ns.clj:23:5: error: first-ns/a-function is called with 0 args but expects 1
src/first_ns.clj:28:11: warning: unused binding x
src/first_ns.clj:34:11: warning: unused binding y
linting took 16ms, errors: 2, warnings: 2
```

**expected behavior**

`clj-kondo` should report arity error when function is invoked with an incorrect arity within a `extend-type` or `extend-protocol` macro.

**Comment by drewverlee** [comment]
Extend-type is referenced here https://github.com/clj-kondo/clj-kondo/blob/2b00d4aa4f685f7e4e3477bb36c8fb0172fa3166/src/clj_kondo/impl/analyzer.clj#L1616

Given the comment, not invalid arity seems to be disabled on purpose.

**Comment by borkdude** [comment]
@drewverlee This is done that way because the analysis for extend-type isn't complete yet. It was low priority since it's not the most used feature in Clojure, but it isn't too hard to support.

**Comment by borkdude** [comment]
Consider upvoting the issue with a thumbs up, then I can prioritize it together with the other issues.

**Comment by borkdude** [comment]
Fixed.
